components/ui/Feed/FeedView.tsx

```diff
@@ -80,156 +54,96 @@ function FeedView({ feed, community = false, header }: FeedViewProps) {
+    if (!firstPost || !community) return undefined;
 
-    showActionSheetWithOptions(
-      {
-        options,
-        cancelButtonIndex,
-        userInterfaceStyle: theme.config.initialColorMode,
-      },
-      (index: number) => {
-        if (index === cancelButtonIndex) return;
+    return {
+      id: firstPost.community.id,
+      name: firstPost.community.name,
+    };
+  }, [firstPost, community]);
 
-        if (index === 0) {
-          feed.setSort("TopDay");
-        } else if (index === 1) {
-          feed.setSort("TopWeek");
-        } else if (index === 5) {
-          feed.setSort("MostComments");
-        } else {
-          feed.setSort(options[index] as SortType);
+  useEffect(() => {
+    navigation.setOptions({
+      headerRight: () => {
+        if (dropdownVisible) {
+          return (
+            <Button
+              title="Cancel"
+              onPress={() => dispatch(setDropdownVisible())}
+            />
+          );
         }
 
-        setSortIcon(SortIconType[index]);
-        flashList?.current?.scrollToOffset({ animated: true, offset: 0 });
-      }
-    );
-  };
-
-  const onEllipsisButtonPress = () => {
-    if (community) {
-      const options = ["Block Community", "Cancel"];
-      const cancelButtonIndex = 1;
-
-      showActionSheetWithOptions(
-        {
-          options,
-          cancelButtonIndex,
-          userInterfaceStyle: theme.config.initialColorMode,
-        },
-        (index: number) => {
-          if (index === cancelButtonIndex) return;
-
-          if (index === 0) {
-            trigger("impactMedium");
-
-            dispatch(
-              showToast({
-                message: `Blocked ${communityName.current}`,
-                duration: 3000,
-                variant: "info",
-              })
-            );
+        return (
+          <HStack space={1}>
+            <FeedSortButton
+              feed={feed}
+              onSortUpdate={() =>
+                flashList?.current?.scrollToOffset({
+                  animated: true,
+                  offset: 0,
+                })
+              }
+            />
+            <FeedOverflowButton
+              feed={feed}
+              community={postCommunity}
+              onPress={() =>
+                flashList?.current?.scrollToOffset({
+                  animated: true,
+                  offset: 0,
+                })
+              }
+            />
+          </HStack>
+        );
+      },
+    });
+  }, [feed, postCommunity, dropdownVisible]);
 
-            lemmyInstance
-              .blockCommunity({
-                auth: lemmyAuthToken,
-                community_id: communityId.current,
-                block: true,
-              })
-              .then();
-          }
-        }
-      );
-    } else {
-      const options = ["All", "Local", "Subscribed", "Cancel"];
-      const cancelButtonIndex = 3;
+  const renderItem = React.useCallback(
```

Comment:
memoizing a few of the callbacks we're passing to FlashList; you typically don't want anonymous functions being passed to a list. It's notorious for impacting performance.